Automattic/wp-calypso
# Posts: Remove PostCountsStore Flux store

This pull request seeks to migrate the remaining single component using `PostCountsStore` Flux store to use Redux post counts state, `<PostsNavigation />`.

__Testing instructions:__

Verify that posts navigation continues to behave as it did previously, with and without Redux state persistence (disable by restarting Calypso with `DISABLE_FEATURES=persist-redux make run`).

Specifically:
- Jetpack sites and all sites posts list do not load counts and always display all statuses
- Post counts update to reflect "Everyone" / "Only Me" selected toggle state
- When shown as select dropdown due to limited space, ensure items reflect correct count
- Statuses are hidden when no counts exist, except Published and Drafts
 - This behaves slightly different than master, specifically that we always show the links for published and drafts even while counts are being loaded (master shows an empty `<SectionNav />`)

Ensure that no remaining references to post counts Flux store remain. I did note a remaining `&meta=counts` query parameter in the sync handler test data, but this was not easily removed (assumed hash generated by query parameters).
